README.md

```diff
@@ -496,6 +495,50 @@ You can include additional monitoring configurations by adding the prefix `monit
 > 
 > It is suggested to turn off Enhanced Failure Monitoring plugin, or to avoid using RDS Proxy endpoints when the plugin is active. 
 
+## AWS Secrets Manager Plugin
+
+The AWS JDBC Driver for MySQL supports usage of database credentials stored in the [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/) through the AWS Secrets Manager Plugin. This plugin is optional and can be enabled with the `connectionPluginFactories` parameter as seen in the [connection plugin manager parameters table](#connection-plugin-manager-parameters). When a user creates a new connection with this plugin enabled, the plugin will retrieve the secret and the connection will be created using those credentials.
+
+The following properties are required for the AWS Secrets Manager Plugin to retrieve secrets.
+
+| Parameter                | Value  | Required | Description                                             | Example          |
+|--------------------------|:------:|:--------:|:--------------------------------------------------------|:-----------------|
+| `secretsManagerSecretId` | String |   Yes    | Set this value to be the secret name or the secret ARN. | `test-secret-id` |
+| `secretsManagerRegion`   | String |   Yes    | Set this value to be the region your secret is in.      | `us-east-1`      |
+
+### Example
+```java
+import java.sql.*;
+import java.util.Properties;
+import com.mysql.cj.jdbc.ha.plugins.AWSSecretsManagerPlugin;
+
+public class AwsIamAuthenticationSample {
```

Comment:
I think class name is a bit misleading. The class is about Secret Manager passwords, not about IAM authentication.